services/gitdiff/gitdiff.go

```diff
@@ -181,7 +181,7 @@ var (
 	removedCodePrefix = []byte(`<span class="removed-code">`)
 	codeTagSuffix     = []byte(`</span>`)
 )
-var addSpanRegex = regexp.MustCompile(`<span [class="[a-z]*]*$`)
+var addSpanRegex = regexp.MustCompile(`<span\s*[class="[a-z]*]*$`)
```

Comment:
Are you sure that this correct?

`[class="[a-z]*]*` seems like a very odd regexp to me

Reply:
Yes it is right unfortunately  -- the only way to properly syntax highlight individual changes within a line (added/removed code class) is to syntax highlight the diffs first so the highlighting itself becomes part of the comparison for lines that have these individual changes highlighted. 

This can cause weird situations where the diff splits on syntax classes in various ways (usually when the change also involves changing the type of code/syntax) and this is to manually fix it. I added some tests for this type of thing starting here which shows how it can end up looking:

 https://github.com/go-gitea/gitea/blob/4cc8697a652a60263149fb794c46df87d19eff74/services/gitdiff/gitdiff_test.go#L45

So it should match any of these types of examples which are all points in which the code might be split for added/removed code sections:

```
<span
<span 
<span class
<span class="p
<span class="nx
```

It is ugly but I didn't seem to find a better way within the diff code to properly syntax highlight these small inline changes since they need to be styled twice.

Reply:
The alternative would be to just not syntax highlight those inline changes but it was requested in original addition of new syntax highlighting

Reply:
But what I mean is that isn't that group simply:

`[a-z="[]*` followed by `]*`

Reply:
Sorry Im not sure I understand, what would be the full suggested change?

Reply:
I'm saying that `[class="[a-z]*]*` is equal to `[a-z="[]*]*`.

It seems like you were actually intending `(class="[a-z]*)?` ?

Reply:
I think I'm still confused and not sure both what is not correct about the current one or why others would be preferred as the current one works as intended.

Sorry I feel slow today 😬

Reply:
https://play.golang.org/p/OU9jEWvG_qQ

Reply:
Ah sorry, it does need to match stuff like "<span c" as well (which is in one of the tests linked above). So while the current one might match things it doesn't need to, those should never be present. I guess in some cases it might be unclear which variation of <span class it might need to match (<span c, <span cl, etc...) 

All this talk makes me dislike this feature even more.

Reply:
OK well I'm absolutely certain that you don't intend trailing `]` or internal `[` characters to match. 

```suggestion
var addSpanRegex = regexp.MustCompile(`<span\s*[a-z="]*$`)
```

If you intend `-` to match add it before the `]` in the regexp.